account_multicurrency_revaluation/model/account.py

```diff
@@ -0,0 +1,320 @@
+                    (state, view) for state, view in action["views"] if view != "form"
+                ]
+            else:
+                action["views"] = form_view
+            action["res_id"] = self.revaluation_origin_line_ids.id
+        else:
+            action = {"type": "ir.actions.act_window_close"}
+        return action
+
+    def action_view_revaluation_created_line(self):
+        self.ensure_one()
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "account.action_account_moves_all"
+        )
+        action["context"] = {}
+        if self.revaluation_created_line_id:
+            form_view = [(self.env.ref("account.view_move_line_form").id, "form")]
+            if "views" in action:
+                action["views"] = form_view + [
+                    (state, view) for state, view in action["views"] if view != "form"
+                ]
+            else:
+                action["views"] = form_view
+            action["res_id"] = self.revaluation_created_line_id.id
+        else:
+            action = {"type": "ir.actions.act_window_close"}
+        return action
+
+
+class AccountAccount(models.Model):
+    _inherit = "account.account"
+
+    currency_revaluation = fields.Boolean(
+        string="Allow Currency Revaluation",
+        compute="_compute_currency_revaluation",
+        store=True,
+        readonly=False,
+    )
+
+    _sql_mapping = {
+        "balance": "COALESCE(SUM(debit),0) - COALESCE(SUM(credit), 0) as balance",
+        "debit": "COALESCE(SUM(debit), 0) as debit",
+        "credit": "COALESCE(SUM(credit), 0) as credit",
+        "foreign_balance": "COALESCE(SUM(amount_currency), 0) as foreign_balance",
+    }
+
+    def init(self):
+        # all receivable, payable, Bank and Cash accounts should
+        # have currency_revaluation True by default
+        res = super().init()
+        accounts = self.env["account.account"].search(
+            [
+                ("account_type", "in", self._get_revaluation_account_types()),
+                ("currency_revaluation", "=", False),
+                ("include_initial_balance", "=", True),
+            ]
+        )
+        accounts.write({"currency_revaluation": True})
+        return res
+
+    def write(self, vals):
+        if (
+            "currency_revaluation" in vals
+            and vals.get("currency_revaluation", False)
+            and any([not x for x in self.mapped("include_initial_balance")])
+        ):
+            raise UserError(
+                self.env._(
+                    "There is an account that you are editing not having the Bring "
+                    "Balance Forward set, the currency revaluation cannot be applied "
+                    "on these accounts: \n\t - %s"
+                )
+                % "\n\t - ".join(
+                    self.filtered(lambda x: not x.include_initial_balance).mapped(
+                        "name"
+                    )
+                )
+            )
```

Comment:
```suggestion
                self.env._(
                    "There is an account that you are editing not having the Bring "
                    "Balance Forward set, the currency revaluation cannot be applied "
                    "on these accounts: \n\t - %s",
                    "\n\t - ".join(
                        self.filtered(lambda x: not x.include_initial_balance).mapped(
                            "name"
                        )
                    )
                )
```
you can read it https://github.com/odoo/odoo/pull/52155

Reply:
if you have time, you can update it